### ISSUE TITLE
Trim does not trim trailing whitespace

### ISSUE THREAD

**Issue by nbauernfeind** [deephaven/deephaven-csv#133]
**Description**

Trailing white space remains even when Trim is set to true.

**Steps to reproduce**

Load the attached csv into a dh-core repl:
```python
from deephaven import read_csv
t = read_csv("test.csv", trim=True, delimiter=",")
```

**Expected results**

I'd expect the `Copy Cell Unformatted` (right-click dropdown menu on a table cell in the deephaven-core web-ui) to demonstrate that all cells are properly trimmed.

**Actual results**

The 2nd and 3rd rows for all three columns have trailing whitespace.

[test.csv](https://github.com/deephaven/deephaven-csv/files/12294402/test.csv)

**Comment by nbauernfeind** [comment]
@kosak it looks like we need to add trim support to `processUnquotedMode`.

```java
    public boolean grabNext(final ByteSlice dest, final MutableBoolean lastInRow)
            throws CsvReaderException {
        spillBuffer.clear();
        startOffset = offset;

        if (ignoreSurroundingSpaces) {
            skipWhitespace();
        }
        if (!tryEnsureMore()) {
            return false;
        }

        // Is first char the quote char?
        if (buffer[offset] == quoteChar) {
            ++offset;
            processQuotedMode(dest, lastInRow);
            if (trim) {
                trimWhitespace(dest);
            }
        } else {
            processUnquotedMode(dest, lastInRow);
        }
        return true;
    }
```

It seem that this ignored leading whitespace:
```java
        if (ignoreSurroundingSpaces) {
            skipWhitespace();
        }
```

However, in `processUnquotedMode(dest, lastInRow);` we have no code path to drop the trailing whitespace.

**Comment by kosak** [comment]
Hi, thanks for the bug report. The issue has been fixed in Deephaven Core 0.27.1 and Deephaven CSV 0.11.0
